velox/tool/trace/QueryReplayer.cpp

```diff
@@ -41,38 +45,68 @@ void init() {
   core::PlanNode::registerSerDe();
   core::ITypedExpr::registerSerDe();
   exec::registerPartitionFunctionSerDe();
+  connector::hive::HiveTableHandle::registerSerDe();
+  connector::hive::LocationHandle::registerSerDe();
+  connector::hive::HiveColumnHandle::registerSerDe();
+  connector::hive::HiveInsertTableHandle::registerSerDe();
+  if (!isRegisteredVectorSerde()) {
+    serializer::presto::PrestoVectorSerde::registerVectorSerde();
+  }
+  // TODO: make it configurable.
+  const auto ioExecutor = std::make_unique<folly::IOThreadPoolExecutor>(3);
+  const auto hiveConnector =
+      connector::getConnectorFactory("hive")->newConnector(
+          "test-hive",
+          std::make_shared<config::ConfigBase>(
+              std::unordered_map<std::string, std::string>()),
+          ioExecutor.get());
+  connector::registerConnector(hiveConnector);
+}
+
+std::unique_ptr<tool::trace::OperatorReplayerBase> createReplayer(
+    const std::string& operatorType) {
+  if (operatorType == "TableWriter") {
+    return std::make_unique<tool::trace::TableWriterReplayer>(
+        FLAGS_root,
+        FLAGS_task_id,
+        FLAGS_node_id,
+        FLAGS_pipeline_id,
+        FLAGS_operator_type,
+        FLAGS_table_writer_output_dir);
+  }
+  return nullptr;
 }
 } // namespace
 
 int main(int argc, char** argv) {
   if (argc == 1) {
-    LOG(ERROR) << "\n" << tool::trace::QueryTraceReplayer::usage();
+    LOG(ERROR) << "\n" << tool::trace::OperatorReplayerBase::usage();
     return 1;
   }
 
   gflags::ParseCommandLineFlags(&argc, &argv, true);
   if (FLAGS_usage) {
-    LOG(INFO) << "\n" << tool::trace::QueryTraceReplayer::usage();
+    LOG(INFO) << "\n" << tool::trace::OperatorReplayerBase::usage();
     return 0;
   }
 
   if (FLAGS_root.empty()) {
     LOG(ERROR) << "Root dir is not provided!\n"
-               << tool::trace::QueryTraceReplayer::usage();
-    return 1;
-  }
-
-  if (!FLAGS_summary && !FLAGS_short_summary) {
-    LOG(ERROR) << "Only support to print traced query metadata for now";
+               << tool::trace::OperatorReplayerBase::usage();
     return 1;
   }
 
   init();
-  const auto tool = std::make_unique<tool::trace::QueryTraceReplayer>();
+  const auto replayer = createReplayer(FLAGS_operator_type);
+  VELOX_USER_CHECK_NOT_NULL(
```

Comment:
Shall we throw or add check inside createReplayer? It might be good to keep the nullable check here as well? Thanks!

Reply:
Sure